src/main/java/uk/ac/ebi/ampt2d/metadata/exceptionhandling/ExceptionHandlers.java

```diff
@@ -33,9 +34,18 @@
         return new ResponseEntity(new ErrorMessage(HttpStatus.BAD_REQUEST, ex, ex.getMessage()), HttpStatus.BAD_REQUEST);
     }
 
+    @ExceptionHandler(value = ConstraintViolationException.class)
+    public ResponseEntity<ErrorMessage> handleConstraintViolationException(ConstraintViolationException ex) {
+        if (ex.getErrorCode() == 23505) { // 23505 is a constraint violation error Code for duplicate id or index
+            return new ResponseEntity(new ErrorMessage(HttpStatus.CONFLICT, ex,
+                    "Duplicate id or index provided"), HttpStatus.CONFLICT);
```

Comment:
Why not using the message from the exception? This message doesn't specify the ID or index so it is not very useful.

Reply:
As the exception is returned from database ( internal server error , we are changing to client error ) which returns too much information like index id and stuffs which is too much for the user i guess.

Reply:
Could you paste an example of the returned exception (either here or in JIRA) so we can decide what to display exactly?

Reply:
``` {
  "cause": {
    "cause": {
      "cause": null,
      "message": "Unique index or primary key violation: \"UK_OLRBTSBQD21KAF4CP7RH4RCHN_INDEX_1 ON PUBLIC.TAXONOMY(TAXONOMY_IDENTIFIER) VALUES (1, 1)\"; SQL statement:\ninsert into taxonomy (id, last_modified_date, name, taxonomy_identifier) values (null, ?, ?, ?) [23505-196]"
    },
    "message": "could not execute statement"
  },
  "message": "could not execute statement; SQL [n/a]; constraint [\"UK_OLRBTSBQD21KAF4CP7RH4RCHN_INDEX_1 ON PUBLIC.TAXONOMY(TAXONOMY_IDENTIFIER) VALUES (1, 1)\"; SQL statement:\ninsert into taxonomy (id, last_modified_date, name, taxonomy_identifier) values (null, ?, ?, ?) [23505-196]]; nested exception is org.hibernate.exception.ConstraintViolationException: could not execute statement"
} ```